components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationResultCache.java

```diff
@@ -109,11 +110,15 @@ public AuthenticationResultCacheEntry getValueFromCache(AuthenticationResultCach
 
     private boolean isCacheEntryExpired(AuthenticationResultCacheEntry entry) {
 
+        if (entry.getResult().getProperty(FrameworkConstants.CREATED_TIMESTAMP) == null) {
+            log.warn("Cache entry does not have a created timestamp.");
+            return false;
+        }
         String createdTimestamp = entry.getResult().getProperty(FrameworkConstants.CREATED_TIMESTAMP).toString();
-        if (createdTimestamp != null &&
+        if (StringUtils.isNotBlank(createdTimestamp) &&
                 (FrameworkUtils.getCurrentStandardNano() >
                     entry.getValidityPeriod() + Long.parseLong(createdTimestamp) * 1000000)) {
-            log.debug("Authentication result cache is expired");
+            log.warn("Authentication result cache is expired");
```

Comment:
Do we need to change this to a warn log?

Reply:
This log is required to check if a authentication is failed due to session being expired